dadi/lib/view/index.js

```diff
@@ -28,40 +27,48 @@ View.prototype.setData = function (data) {
 }
 
 View.prototype.render = function (done) {
-  if (this.json) {
-    // Return the raw data
-    return done(null, this.data)
-  } else {
-    let templateData = _.extend(
-      {
-        host: this.page.hostKey
-      },
-      this.data
-    )
+  // Return the raw data
+  if (this.json) return done(null, this.data)
 
-    this.template
-      .render(templateData, {
-        keepWhitespace: this.page.keepWhitespace
-      })
-      .then(output => {
-        let err = null
+  // Send the templated page
+  let templateData = Object.assign(this.data, {
```

Comment:
Could this be a `const`?

Reply:
Also, to avoid mutating `this.data`, you could do `Object.assign({}, this.data, (...))`.